compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1226,10 +1245,24 @@ pub fn report_ice(info: &panic::PanicInfo<'_>, bug_report_url: &str) {
     }
 
     handler.emit_note(session_diagnostics::IceBugReport { bug_report_url });
-    handler.emit_note(session_diagnostics::IceVersion {
-        version: util::version_str!().unwrap_or("unknown_version"),
-        triple: config::host_triple(),
-    });
+    let mut file = ICE_PATH
+        .as_ref()
+        .filter(|path| path.exists())
+        .and_then(|p| crate::fs::File::options().append(true).open(&p).ok().map(|f| (f, p)));
+
+    let version = util::version_str!().unwrap_or("unknown_version");
+    let triple = config::host_triple();
+
+    static FIRST_PANIC: AtomicBool = AtomicBool::new(true);
+
+    if let Some((ref mut file, path)) = file {
+        handler.emit_note(session_diagnostics::IcePath { path: path.display().to_string() });
+        if FIRST_PANIC.swap(false, Ordering::SeqCst) {
+            let _ = write!(file, "\n\nrustc version: {version}\nplatform: {triple}");
+        }
+    } else {
+        handler.emit_note(session_diagnostics::IceVersion { version, triple });
+    }
 
```

Comment:
Please give this code a few comments and/or split it out into (one or more) well-named functions.